src/vdom/diff.js

```diff
@@ -160,32 +160,24 @@ function idiff(dom, vnode, context, mountAll, componentRoot) {
  *	@param {Array} vchildren		Array of VNodes to compare to `dom.childNodes`
  *	@param {Object} context			Implicitly descendant context object (from most recent `getChildContext()`)
  *	@param {Boolean} mountAll
- *	@param {Boolean} isHydrating	If `true`, consumes externally created elements similar to hydration
  */
-function innerDiffNode(dom, vchildren, context, mountAll, isHydrating) {
+function innerDiffNode(dom, vchildren, context, mountAll) {
 	let originalChildren = dom.childNodes,
-		children = [],
 		keyed = {},
 		keyedLen = 0,
-		min = 0,
 		len = originalChildren.length,
-		childrenLen = 0,
 		vlen = vchildren ? vchildren.length : 0,
-		j, c, f, vchild, child;
+		f, vchild, child;
 
 	// Build up a map of keyed children and an Array of unkeyed children:
 	if (len!==0) {
 		for (let i=0; i<len; i++) {
 			let child = originalChildren[i],
 				props = child[ATTR_KEY],
 				key = vlen && props ? child._component ? child._component.__key : props.key : null;
-			if (key!=null) {
-				keyedLen++;
-				keyed[key] = child;
-			}
-			else if (props || (child.splitText!==undefined ? (isHydrating ? child.nodeValue.trim() : true) : isHydrating)) {
-				children[childrenLen++] = child;
-			}
+			key = key || ( ELT_KEY_PREFIX + i );
```

Comment:
If `key` was explicitly assigned as `0` (or empty string), this would reassign to the prefixed index. Is that intended? Guessing this is what would have been previously respected by `key!=null` defaulting only in the case of `null` and `undefined`.

Reply:
Good catch